enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_old`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`\
+   Action = Existing tag for AWS resource is updated to reflect new value.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about user-defined tag being marked for deletion.
+
+### Delete tags scenarios
+User-defined tags are deleted when the user sets the user-defined tag value to empty string in `.spec.platformSpec.aws.resourceTags`.
+Also refer to `Precedence` scenario where delete of user-defined tag is not allowed to delete.\
+For example,\
+Existing tag for AWS resource = `key_infra1 = value_old`\
+New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`\
+Action = Existing tag for AWS resource is deleted.\
+Final tag set to AWS resource = deleted\
+Event action = An event is generated to notify user about the action status (success/failure) to delete tags for the AWS resource.
+
+### Precedence scenarios
+1) User-defined tags on local objects MUST continue to take precedence during create and update. User-defined tags found on local objects must not be deleted by methods described in `Delete tags scenarios`.
+   A warning is generated to inform the user about action not being applied on the list of user tags.
+
+   For example,\
+   Existing tag in local object like machine CRD = `key_infra1 = custom_value`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value1`\
+   Action = The tag for AWS resource is maintained to `key_infra1 = custom_value`.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about list of user-defined tags on which action is not applicable.
+
+2) `.spec.platformSpec.aws` take precedence over `.status.platformStatus.aws`. User-defined tags must be merged from `.status.platformStatus.aws` and set to AWS resource.
+   On delete, user-defined must also be removed from `.status.platformStatus.aws`, if present.
+
+### Caveats
+1) User updates the user-defined tag from using external tools when there is an entry in `.spec.platformSpec.aws.resourceTags`
+   The user-defined tag which is updated from spec, will be reconciled by operators to set value from `.spec.platformSpec.aws.resourceTags`.
+   The user-defined tag will be overwritten with value from `.spec.platformSpec.aws.resourceTags` when there is an update to `.spec.platformSpec.aws.resourceTags`.
```

Comment:
Is the operator (1) not expected to restore the value of the tag until there is a separate change to `.spec.platformSpec.aws.resourceTags` or (2) not allowed to restore the value of the tag until such a change is made?

Reply:
I commented on this before, but it seems to have disappeared. The wording of an update to the tags doesn't make sense here, if we are reconciling resources, eg Machines, we will update the tags whenever the Machine is reconciled, can we reword this to make it clearer

Reply:
Rephased it.